### ISSUE TITLE
Improve hashCode & equals for JFace wrappers

### ISSUE THREAD

**Issue by IanMayo** [debrief/limpet#180]
We needed good quality hashCode & equals methods in order for the JFace TreeViewer to remember folder state during a refresh operation.

But, @dinkoivanov pointed out that:

```
 Since a model object might appear in multiple places in the tree,  
 with the current approach it won't be possible to select certain 
 nodes in the tree (JFace viewer will pickup the first appearance 
 I believe). 
```

We should ensure this is handled correctly.

**Comment by pecko** [comment]
@IanMayo 

We could use UUID that you added for drag and drop. Each object has its own UUID.

**Comment by IanMayo** [comment]
@dinkoivanov indicated that a model object may appear at numerous locations in the Tree - so each shares the same UUID.  That appears to be why we should use a HashCode that relates to both the model object and the JFace wrapper

**Comment by pecko** [comment]
Are you sure? If I understand correctly, they have different parent, so they have to be different objects.

**Comment by IanMayo** [comment]
No, some datasets show their precedents and dependents.  When this happens, the child element  (descendent) of a dataset will already be visible elsewhere in the tree view.

**Comment by pecko** [comment]
I don't sure what you used as equals/hashCode, but UUID is logical because it already exists and it is certainly unique. If one parent can have the same children then you just create equals/hashCode based on UUID and parent.equals/hashCode.
